AutomatedTesting/Gem/PythonTests/Terrain/EditorScripts/TerrainSystem_VegetationSpawnsOnTerrainSurfaces.py

```diff
@@ -0,0 +1,217 @@
+    
+    general.idle_wait_frames(1)
+
+    box_height = 20.0
+    box_dimensions = math.Vector3(20.0, 20.0, box_height)
+
+    # Add Terrain Rendering
+    hydra.add_level_component("Terrain World")
+    hydra.add_level_component("Terrain World Renderer")
+
+    # Create two terrain entities at adjoining positions
+    terrain_entity_1 = create_entity_at("Terrain1", ["Terrain Layer Spawner", "Axis Aligned Box Shape", "Terrain Height Gradient List", "Terrain Surface Gradient List"], 0.0, 0.0, box_height/2.0)
+    terrain_entity_1.get_set_test(1, "Axis Aligned Box Shape|Box Configuration|Dimensions", box_dimensions)
+
+    terrain_entity_2 = create_entity_at("Terrain2", ["Terrain Layer Spawner", "Axis Aligned Box Shape", "Terrain Height Gradient List", "Terrain Surface Gradient List"], 20.0, 0.0, box_height/2.0)
+    terrain_entity_2.get_set_test(1, "Axis Aligned Box Shape|Box Configuration|Dimensions", box_dimensions)
+
+    # Create two gradient entities.
+    gradient_value_1 = 0.25
+    gradient_value_2 = 0.5
+
+    gradient_entity_1 = create_entity_at("Gradient1", ["Constant Gradient"], 0.0, 0.0, 0.0)
+    gradient_entity_1.get_set_test(0, "Configuration|Value", gradient_value_1)
+
+    gradient_entity_2 = create_entity_at("Gradient2", ["Constant Gradient"], 0.0, 0.0, 0.0)
+    gradient_entity_2.get_set_test(0, "Configuration|Value", gradient_value_2)
+
+    # Assign mappings to the gradient surface lists, but don't assign SurfaceTags yet.
+    surface_tags: dict = {"test_tag2": 1797181423, "test_tag3": 471465849, "terrain": 3363197873}
+
+    mapping = terrain.TerrainSurfaceGradientMapping()
+    mapping.gradientEntityId = gradient_entity_1.id
+    pte = hydra.get_property_tree(terrain_entity_1.components[3])
+    pte.add_container_item("Configuration|Gradient to Surface Mappings", 0, mapping)
+
+    mapping = terrain.TerrainSurfaceGradientMapping()
+    mapping.gradientEntityId = gradient_entity_1.id
+    pte = hydra.get_property_tree(terrain_entity_2.components[3])
+    pte.add_container_item("Configuration|Gradient to Surface Mappings", 0, mapping)
+
+    # create a vegetation entity that overlaps both terrain entities.
+    vegetation_entity = create_entity_at("Vegetation", ["Vegetation Layer Spawner", "Axis Aligned Box Shape", "Vegetation Asset List", "Vegetation Surface Mask Filter"], 10.0, 0.0, box_height/2.0)
+    vegetation_entity.get_set_test(1, "Axis Aligned Box Shape|Box Configuration|Dimensions", box_dimensions)
+
+    # Set the vegetation area to a Dynamic Slice spawner with a specific slice asset selected.
+    dynamic_slice_spawner = vegetation.DynamicSliceInstanceSpawner()
+    dynamic_slice_spawner.SetSliceAssetPath(os.path.join("Slices", "PinkFlower.dynamicslice"))
+    descriptor = hydra.get_component_property_value(vegetation_entity.components[2], 'Configuration|Embedded Assets|[0]')
+    descriptor.spawner = dynamic_slice_spawner
+    vegetation_entity.get_set_test(2, "Configuration|Embedded Assets|[0]", descriptor)
```

Comment:
This should use a PrefabInstanceSpawner() instead and point at a .spawnable file, so that it's one less thing to convert over to prefabs.

Reply:
Done